src/fluree/db/db/json_ld.cljc

```diff
@@ -28,6 +28,14 @@
    const/iri-modify {:root? true}})
 
 
+(defn lookup-id
+  "Returns subject id or nil if no match."
+  [db iri]
+  (go-try
+    (some-> (<? (query-range/index-range db :post = [const/$xsd:anyURI iri]))
+            first
+            flake/s)))
```

Comment:
I know this is an extraction of a previously existing function and it probably doesn't make a difference in the grand scheme of things, but we can pass `(map flake/s)` in the `:flake-xf` option of `index-range` to do this during index retrieval instead of after the fact.

Reply:
For some reason rewriting it caused a test failure that I'm utterly perplexed by. I'm going to merge it as-is, but I'll address it on the other branch.

Reply:
no problem. don't stress too much about the test failure if it still happens in the other branch for now we'll figure it out when the dust settles